### PR TITLE
Introduce Dockerfile of Jupyter editor

### PR DESCRIPTION
**Pull request by skabashnyuk**
Introduce Dockerfile of Jupyter editor
che-plugin.yaml moved to  https://github.com/eclipse/che-plugin-registry/blob/master/v3/plugins/ws-skeleton/jupyter/1.0.0/meta.yaml

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+# Enable a more liberal Content-Security-Policy so that we can display Jupyter
+# in an iframe.
+RUN echo "c.NotebookApp.tornado_settings = {" >> /etc/jupyter/jupyter_notebook_config.py && \
+       echo "    'headers': {" >> /etc/jupyter/jupyter_notebook_config.py && \
+       echo "        'Content-Security-Policy': \"frame-ancestors 'self' *\"" >> /etc/jupyter/jupyter_notebook_config.py && \
+       echo "    }" >> /etc/jupyter/jupyter_notebook_config.py && \
+       echo "}" >> /etc/jupyter/jupyter_notebook_config.py
+
+# Override command to startup Jupyter notebook. The original is wrapped
```
Comment (sleshchenko) [inline Dockerfile:67]: Looks like it's a comment for last command `CMD` but not for `RUN`

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      io.openshift.s2i.scripts-url="image:///opt/app-root/s2i/bin"
+
+# Copy in S2I builder scripts for installing Python packages and copying
+# in of notebooks and data files.
+
+COPY s2i /opt/app-root/s2i
+
+# Adjust permissions on home directory so writable by group root.
+
+RUN chgrp -Rf root /home/$NB_USER && chmod -Rf g+w /home/$NB_USER
```
Comment (sleshchenko) [inline Dockerfile:31]: Can we merge this RUN and the next to make one more layer less?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+# process will reject the builder image as can't tell if user name
+# really maps to user ID for root.
+
+USER 1000
+
+# Set up Jupyter Notebook config
+ENV CONFIG /home/jovyan/.jupyter/jupyter_notebook_config.py
+ENV CONFIG_IPYTHON /home/jovyan/.ipython/profile_default/ipython_config.py
+
+RUN bash -c 'source activate base && rm /home/jovyan/.jupyter/jupyter_notebook_config.py && jupyter notebook --generate-config --allow-root && \
```
Comment (sleshchenko) [inline Dockerfile:47]: Please consider merging of multiple `RUN` in one to avoid additional image layers.

**Review by sleshchenko — APPROVED** [review]
LGTM

**Comment by skabashnyuk** [comment]
@sleshchenko thanks for the review. I'll try to address your comments in next optimizations of this image. Now I would like to merge it as is because I'm scared a lot that it will not work.

**Pull request merged**